README.md

```diff
@@ -176,65 +176,92 @@ the new state.
-    >>>     # Store the credentials state somewhere, e.g. a cache
-    >>>     cred_state = credentials.state
-    >>>     caches['mycache'].set('xero_creds', cred_state)
-    >>>     return HttpResponseRedirect(authorization_url)
-    ...
-    ```
-    `credentials.auth_state` should be stored by your app (e.g. in the database
-    or cache) as it will be used to complete the authorization during the callback.
-    The simplest way is to dump the whole credentials object by storing `credentials.state`
-
- 3) After authorization the user will be redirected to the callback URI
-    provided (e.g. `https://example.com/oauth/xero/callback`) along with a querystring
-    containing the authentication secret. Pass the full URI including querystring to
-    `verify()`, using the credentials constructed earlier:
-    ```python
-    >>> def complete_xero_auth_view(request):
-    >>>     # Create the credentials object from wherever you stored the state
-    >>>     credentials = OAuth2Credentials(**cred_state)
-    >>>     # Then pass in the full URI the user was directed back to, including
-    >>>     # querystring
-    >>>     auth_secret = request.get_raw_uri()
-    >>>     credentials.verify(auth_secret)
+    >>>     # Get the full redirect uri from the request including querystring
+    >>>     # e.g. request_uri = 'https://mysite.com/oauth/xero/callback/?code=0123456789&scope=openid%20profile&state=87784234sdf5ds8ad546a8sd545ss6'
+    >>>
+    >>>     credentials.verify(request_uri)
     ```
-    A token will be fetched from Xero and saved as `credentials.token`. If the credentials
-    object needs to be created again either dump the whole object using:
+    A token will be fetched from Xero and saved as `credentials.token`. If the 
+    credentials object needs to be created again either dump the whole object 
+    using:
     ```python
     >>> cred_state = credentials.state
     >>> ...
     >>> new_creds = OAuth2Credentials(**cred_state)
     ```
-    or just use the client_id, client_secret and token:
+    or just use the client_id, client_secret and token (and possibly scopes):
     ```python
     >>> token = credentials.token
     >>> ...
     >>> new_creds = OAuth2Credentials(client_id, client_secret, token=token)
     ```
+  
+    As an example steps 2-3 might be done as follows in Django:
+
+    ```python
+    from django.http import HttpResponseRedirect
+    from django.core.cache import caches
+    from xero.auth import OAuth2Credentials
+    
+    def start_xero_auth_view(request):
+        credentials = OAuth2Credentials(
+            client_id, client_secret, callback_uri=callback_uri, scope=scope
+        )
+        authorization_url = credentials.generate_url()
+        caches['mycache'].set('xero_creds', credentials.state)
+        return HttpResponseRedirect(authorization_url)
+    
+    def process_callback_view(request):
+        credentials = OAuth2Credentials(**cred_state)
```

Comment:
Need to get the cred_state from the cache, will put this in with my next set of changes.